lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -962,6 +1027,74 @@ static LogicalResult buildAllocOp(ComponentLoweringState &componentState,
                     IntegerAttr::get(rewriter.getI1Type(), llvm::APInt(1, 1)));
   componentState.registerMemoryInterface(allocOp.getResult(),
                                          calyx::MemoryInterface(memoryOp));
+
+  bool isFloat = !memtype.getElementType().isInteger();
+
+  auto shape = allocOp.getType().getShape();
+  std::vector<int> dimensions;
+  int totalSize = 1;
+  for (auto dim : shape) {
+    totalSize *= dim;
+    dimensions.push_back(dim);
+  }
+
+  std::vector<double> flattenedVals(totalSize, 0);
+
+  // Helper function to get the correct indices
+  auto getIndices = [&dimensions](int flatIndex) {
+    std::vector<int> indices(dimensions.size(), 0);
```

Comment:
It seems a bit overkill to allocate a new vector O(flattenedVals.size()) times. Do you think it would be worth it just to inline this code, and create 
`std::vector<int> indices(dimensions.size(), 0);` outside of the loop where this is called?

Reply:
not used anymore, resolving conversation